src/pages/MovieDetailPage.js

```diff
@@ -1,142 +1,155 @@
-    const [showModal, setShowModal] = useState(false);
-    const [ids, setIds] = useState([]);
+  const { id } = useParams();
+  const [loading, setLoading] = useState(false);
+  const [movie, setMovie] = useState({});
+  const [hasError, setHasError] = useState(false);
+  const [recommendMovies, setRecommendMovies] = useState([]);
+  const [similarMovies, setSimilarMovies] = useState([]);
+  const [reviews, setReviews] = useState([]);
+  const [showModal, setShowModal] = useState(false);
+  const [ids, setIds] = useState([]);
+  const [show, setShow] = useState(false);
+  const [key, setKey] = useState("");
 
-    useEffect(() => {
-        if(!id) {
-            return;
-        }
-        async function fetchData() {
-          setLoading(true);
-          try {
-            const url = `${API_URL}/movie/${id}?api_key=${API_KEY}&language=en-US`;
-            const response = await fetch(url);
-            const data = await response.json();
-            console.log(data);
-            setMovie(data);
-          } catch (error) {
-            setHasError(true);
-          }
-          setLoading(false);
-        };
-        fetchData();
-      }, [id]);
-
-    useEffect(() => {
-      if(!id) {
-          return;
+  useEffect(() => {
+    if (!id) {
+      return;
+    }
+    async function fetchData() {
+      setLoading(true);
+      try {
+        const url = `${API_URL}/movie/${id}?api_key=${API_KEY}&language=en-US`;
+        const response = await fetch(url);
+        const data = await response.json();
+        console.log(data);
+        setMovie(data);
+      } catch (error) {
+        setHasError(true);
       }
-      async function fetchIds() {
-        try {
-          const url = `${API_URL}/movie/${id}/videos?api_key=${API_KEY}&language=en-US`;
-          const response = await fetch(url);
-          const data = await response.json();
-          setIds(data.results)
-        } catch (error) {
-          setHasError(true);
-        }
-      };
-      fetchIds();
-    }, [id]);
+      setLoading(false);
+    }
+    fetchData();
+  }, [id]);
 
-      useEffect(() => {
-        if(!id) {
-            return;
-        }
-        async function fetchRecommendMovie() {
-            const url = `${API_URL}/movie/${id}/recommendations?api_key=${API_KEY}&language=en-US&page=1`;
-            const response = await fetch(url);
-            const data = await response.json();
-            setRecommendMovies(data.results)
-        };
-        fetchRecommendMovie();
-      }, [id]);
+  useEffect(() => {
+    if (!id) {
+      return;
+    }
+    async function fetchIds() {
+      try {
+        const url = `${API_URL}/movie/${id}/videos?api_key=${API_KEY}&language=en-US`;
+        const response = await fetch(url);
+        const data = await response.json();
+        setIds(data.results);
+        console.log(data.results);
+        const video = data.results[Math.floor(Math.random() * ids.length)];
```

Comment:
Yeah. Random is the best way. I am really love the way choosing trailer like you doing.